test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,264 @@
+	peerCmd := cmdExec("wsl", "--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./rancher-desktop-host-resolver", "vsock-peer")
+	peerCmd.Dir = tmpDir
+	err = peerCmd.Start()
+	require.NoError(t, err, "Starting host-resolver peer process faild")
+
+	t.Log("Starting host-resolver host process")
+	resolverExecPath := fmt.Sprintf("%v/rancher-desktop-host-resolver.exe", tmpDir)
+	hostCmd := cmdExec(resolverExecPath, "vsock-host",
+		"--upstream-servers", fmt.Sprintf("[%v]", testSrvAddr))
+	hostCmd.Dir = tmpDir
+	err = hostCmd.Start()
+	require.NoError(t, err, "Starting host-resolver host process faild")
+
+	t.Log("Building DNS hammer binary")
+	err = buildBinaries("../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building dnsHammer")
+
+	err = copyTestData("../testdata/test-300.csv", fmt.Sprintf("%s/test-300.csv", tmpDir))
+	require.NoError(t, err, "Failed copying test data file")
+
+	t.Log("Confirming host-resolver peer process is up")
+	peerCmdTimeout := time.Second * 10
+	confirm(func() bool { //nolint:errcheck // we don't care about the error
+		p, _ := os.FindProcess(peerCmd.Process.Pid)
+		return p.Pid == peerCmd.Process.Pid
+	}, tryInterval, peerCmdTimeout)
+
+	t.Logf("Running dns hammer test process in wsl [%v]", wslDistroName)
+	dnsSrvAddr := net.JoinHostPort(testSrvAddr, dnsPort)
+	runTestCmd := cmdExec(
+		"wsl",
+		"--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./test", "dnshammer",
+		"--server-address", dnsSrvAddr,
+		"--rr-type", "A=test-300.csv")
+	runTestCmd.Dir = tmpDir
+	err = runTestCmd.Run()
+	require.NoError(t, err, "Running dns hammer against the peer process faild")
+
+	_ = hostCmd.Process.Kill()
+	_ = peerCmd.Process.Kill()
+	_ = runTestCmd.Process.Kill()
+}
+
+// TODO (Nino-K): maybe this can be replaced by CI
+func buildBinaries(path, goos, tmpDir string) error {
+	buildCmd := exec.Command("go", "build", "-o", tmpDir, path)
+	buildCmd.Env = os.Environ()
+	buildCmd.Env = append(buildCmd.Env, fmt.Sprintf("GOOS=%s", goos), "GOARCH=amd64")
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+
+	return buildCmd.Run()
+}
+
+func cmdRunWithOutput(command string, args ...string) (string, error) {
+	var outBuf, errBuf strings.Builder
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	err := cmd.Run()
+	if err != nil {
+		return errBuf.String(), err
+	}
+	return outBuf.String(), nil
+}
+
+func cmdExec(command string, args ...string) *exec.Cmd {
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func copyTestData(src, dst string) error {
+	bytesRead, err := os.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(dst, bytesRead, 0600)
+}
+
+func cacheExsitingDNSAddrs(adapterAddrs []*winipcfg.IPAdapterAddresses) (map[string][]netip.Addr, error) {
+	var adapterAddrsCopy = make([]*winipcfg.IPAdapterAddresses, len(adapterAddrs))
+	copy(adapterAddrsCopy, adapterAddrs)
```

Comment:
Huh, does this copy the pointer or the contents? (Given that both src and dest are slices of pointers)

Reply:
Actually, I'm not even sure what the point of copy was; we never modify `adapterAddrs`, and `adapterAddrsCopy` goes out of scope when the function returns…

Reply:
Yes, I agree. It no longer makes sense, got left behind from refactoring.